src/sdk/service/iam-service.ts

```diff
@@ -0,0 +1,178 @@
+import { CompanyInventory, InventoryEntity, Permission, Policy, PolicyBuilder, Role, User } from '../model';
+import { PermissionService } from './permission-service';
+import { PermissionType } from '../model/json';
+
+/**
+ * IamService
+ */
+export class IamService {
+  private static instance: IamService;
+  private _permissionService: PermissionService;
+
+  private constructor() {
+    this._permissionService = PermissionService.getInstance();
+  }
+
+  /**
+   * Get an instance of IamService. This is a singleton implementation.
+   * @returns {IamService}
+   */
+  static getInstance() {
+    if (!IamService.instance) {
+      IamService.instance = new IamService();
+    }
+    return IamService.instance;
+  }
+
+  /**
+   * Get the permission service.
+   * @returns {PermissionService}
+   */
+  get permissionService(): PermissionService {
```

Comment:
is there a good reason to have this here? Seems more natural and less confusing to just use PermissionService.getInstance() wherever we need it

Reply:
Yeah ok. I can do this. np